src/components/TrapFocus/TrapFocus.tsx

```diff
@@ -14,14 +14,36 @@ export interface Props {
   children?: React.ReactNode;
 }
 
-export default class TrapFocus extends React.PureComponent<Props, never> {
-  private focusTrapWrapper: HTMLElement | null = null;
+export interface State {
+  shouldFocusSelf: boolean | undefined;
+}
+
+export default class TrapFocus extends React.PureComponent<Props, State> {
+  state = {
+    shouldFocusSelf: undefined,
```

Comment:
You shouldn't have to initialize `shouldFocusSelf` when optional (`undefined`).

Reply:
During the components first render the content of the components will not be focused. This is a little hacky way of saying once the component has rendered, render the component again but don't steal focus if the active element is contained inside itself.